test/functional/core/remote_spec.lua

```diff
@@ -0,0 +1,56 @@
+local helpers = require('test.functional.helpers')(after_each)
+
+local eq = helpers.eq
+local expect = helpers.expect
+local funcs = helpers.funcs
+local new_argv = helpers.new_argv
+local set_session = helpers.set_session
+local spawn = helpers.spawn
+local tmpname = helpers.tmpname
+local write_file = helpers.write_file
+
+describe('Remote', function()
+  describe('file given', function()
+    local server, fname
+    local contents = 'The call is coming from outside the process'
+
+    before_each(function()
+      fname = tmpname()
+      write_file(fname, contents)
+      server = spawn(new_argv(), true)
+    end)
+
+    after_each(function()
+      server:close()
+    end)
+
+    function run_remote(...)
+      set_session(server)
+      local addr = funcs.serverlist()[1]
+      local client_argv = new_argv({args={'--server', addr, ...}})
+
+      -- Create an nvim instance just to run the remote-invoking nvim. We want
+      -- to wait for the remote instance to exit and calling jobwait blocks
+      -- the event loop. If the server event loop is blocked, it can't process
+      -- our incoming --remote calls.
+      local client_starter = spawn(new_argv(), false, nil, true)
```

Comment:
I'd love some input on if there's a way to spawn a process and wait for it to exit without requiring an nvim instance just for that.

I was able to use native.wait from lua-client, but that won't work on Windows whereas I'm guessing jobwaitwill?